src/tls.rs

```diff
@@ -0,0 +1,194 @@
+//! Connection helper.
+use tokio::io::{AsyncRead, AsyncWrite};
+
+use tungstenite::{
+    client::uri_mode, error::Error, handshake::client::Response, protocol::WebSocketConfig,
+};
+
+use crate::{client_async_with_config, domain, IntoClientRequest, WebSocketStream};
+
+pub use crate::stream::MaybeTlsStream;
```

Comment:
Hmm, MaybeTlsStream seems only available under `stream` flag, but here it is imported unconditionally - I wonder whether this compiles with `*-tls` + `!stream`...?

Reply:
This is save as you can't disable the `stream` feature and have a `*-tls` feature at the same time.
The `*-tls` features require the `stream` feature so the developer can't disable it.

In addition I didn't get any errors when running `cargo hack check --feature-powerset --no-dev-deps` which tests all possible feature flag combinations.

Reply:
> This is save as you can't disable the stream feature and have a *-tls feature at the same time.
The *-tls features require the stream feature so the developer can't disable it.

Oh, that's the missing piece that didn't occur to me, thanks for explaining. It seemed strange to me because I know you're running `cargo hack`.